pgmpy/models/BayesianNetwork.py

```diff
@@ -641,7 +646,14 @@ def predict(self, data, stochastic=False, n_jobs=-1):
             raise ValueError("Data has variables which are not in the model")
 
         missing_variables = set(self.nodes()) - set(data.columns)
-        model_inference = VariableElimination(self)
+
+        if algo is None:
+            algo = VariableElimination
+        else:
+            if not issubclass(algo, Inference):
+                raise TypeError("Algorithm should be a valid pgmpy inference method.")
```

Comment:
You can modify the error message to be a bit more informative by adding the type detected in the code. Something like adding "Got {type(algo)}" at the end of the message.